pdns/dnsdist.cc

```diff
@@ -748,6 +749,22 @@ bool processQuery(LocalStateHolder<NetmaskTree<DynBlock> >& localDynNMGBlock,
     g_rings.queryRing.push_back({now,*dq.remote,*dq.qname,dq.len,dq.qtype,*dq.dh});
   }
 
+  if(g_qcount.enabled) {
+    WriteLock wl(&g_qcount.queryLock);
+    string qname = (*dq.qname).toString(".", false);
+    bool countQuery{true};
+    if(g_qcount.filter) {
+      std::tie (countQuery, qname) = g_qcount.filter(dq);
```

Comment:
I think you should acquire the Lua mutex here with something like `std::lock_guard<std::mutex> lock(g_luamutex);`